.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  schedule:
+    # Run weekly to keep Rust toolchain changes fresh
+    - cron: '0 0 * * 1'
+
+jobs:
+  multiple_toolchains:
+    name: Stable and Beta tasks
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install ${{ matrix.rust }} toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+            toolchain: ${{ matrix.rust }}
+            components: clippy, rustfmt
+      - name: Check formatting
+        run: cargo fmt --all --check
+      - name: Check clippy
+        run: cargo clippy -- -D warnings
```

Comment:
Personally, I think clippy is overlay pedantic sometimes. I think running it from time to time suffices rather than breaking CI.

Reply:
> Personally, I think clippy is overlay pedantic sometimes.

I agree! But I think in that case it's best to `#[allow()]` and `#![allow()]` the specific lints and lint groups that you disagree with so that `cargo clippy` still stays clean. Selectively ignoring lints makes it easier to start missing new lints that start popping up in the future